pkg/eksd/upgrader.go

```diff
@@ -23,34 +25,45 @@ func NewUpgrader(client EksdInstallerClient, reader Reader, opts ...UpgraderOpt)
 	}
 }
 
-func (u *Upgrader) Upgrade(ctx context.Context, cluster *types.Cluster, currentSpec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
+// Upgrade checks whether upgrading EKS-D components is necessary and, if so, installs the new EKS-D CRDs.
+func (u *Upgrader) Upgrade(ctx context.Context, cluster *types.Cluster, currentManagementComponents, newManagementComponents *cluster.ManagementComponents, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
```

Comment:
Maybe we need to think a bit more here about this
This updates the eks-d CRDs in case new fields or versions have been added.
But that's not "upgrading eks-d", it's not actually changing any cluster component.

I'm fine with the method. But the way we build the change diff I think it's misleading. I think it indicates we are upgrading from one eks-d version to another and we are not

TBH I don't have a solution, it's the first time I think about this confusion. Maybe we can sync offline and see if you have some ideas

Reply:
Yeah, we'll continue this offline

Reply:
Discussed offline, we'll change the messaging in the logs, but also change this method to only return an error. This will make it more clear that it is not actually upgrading eks-d from one version to another, but updating the crds when necessary.